bin/transform-genbank

```diff
@@ -0,0 +1,288 @@
+        (r'^USAWA-', 'USA/WA-'),
+        (r'^HKG.', 'HongKong/'),
+    ]
+
+    # Parse strain name from title to fill in strains that are empty strings
+    genbank_data['strain_from_title'] = genbank_data['title'].apply(parse_strain_from_title)
+    genbank_data.loc[(genbank_data['strain'] == ''), 'strain'] = genbank_data['strain_from_title']
+
+    # Standardize strain names using list of regex replacements
+    for regex, replacement in regex_replacement:
+        genbank_data['strain'] = genbank_data['strain'] \
+            .str.replace(regex, replacement, n=1, case=False)
+
+    # Strip all spaces
+    genbank_data['strain'] = genbank_data['strain'].str.replace(r'\s', '')
+
+    return genbank_data
+
+
+def parse_strain_from_title(title: str) -> str:
+    """
+    Try to parse strain name from the given *title* using regex search.
+    Returns an empty string if not match is found in the *title*.
+    """
+    strain_name_regex = r'[-\w]*/[-\w]*/[-\w]*\s'
+    strain = re.search(strain_name_regex, title)
+    return strain.group(0) if strain else ''
+
+
+def parse_geographic_columns(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Expands string found in the column named `location` in the given
+    *genbank_data* DataFrame, creating 3 new columns. Returns the modified
+    DataFrame.
+
+    Expected formats of the location string are:
+        * "country"
+        * "country: division"
+        * "country: division, location"
+    """
+    # Create dict of US state codes and their full names
+    us_states = pd.read_csv(base / 'source-data/us-state-codes.tsv', header=None, sep='\t', comment="#")
+    us_states = dict(zip(us_states[0], us_states[1]))
+
+    geographic_data = genbank_data['location'].str.split(':\s*', expand=True)
+    geographic_data[0] = geographic_data[0].str.strip()
+
+    divisions = []
+    locations = []
+    for index, value in geographic_data.iterrows():
+        # Both location and division are not available
+        if pd.isna(value[1]):
+            location = division = None
+        # Only division is available
+        elif ',' not in value[1]:
+            location = None
+            division = value[1]
+        # Location and division are both available
+        elif ',' in value[1]:
+            division,location = value[1].split(',', 1)
+        # Unknown format of location data
+        else:
+            assert False, f"Found unknown format for geographic data: {value}"
+
+        # Special parsing for US locations because the format varies
+        if value[0] == 'USA' and division:
+            # Switch location & division if location is a US state
+            if location and any(location.strip() in s for s in us_states.items()):
+                state = location
+                location = division
+                division = state
+            # Convert US state codes to full names
+            if us_states.get(division.strip()):
+                division = us_states[division.strip()]
```

Comment:
Might be more robust to normalize (internal space collapsing, lowercase, etc) `location` and `division` and/or `us_states` before comparing them in the `if … any(location … in us_states…)` and `if us_states.get(…)` lines.